utilities/tools/Test-NamePrefixAvailability.ps1

```diff
@@ -56,127 +64,164 @@ function Test-NamePrefixAvailability {
 
-        # Construct Token Configuration Input
-        $tokenConfiguration = @{
-            FilePathList = $parameterFiles
-            Tokens       = @{
-                'namePrefix' = $namePrefix
-            }
-            TokenPrefix  = $GlobalVariablesObject | Select-Object -ExpandProperty tokenPrefix
-            TokenSuffix  = $GlobalVariablesObject | Select-Object -ExpandProperty tokenSuffix
+        $parameterFiles = (Get-ChildItem -Path $repoRoot -Recurse -Filter 'deploy.test.bicep').FullName | Where-Object {
+            Test-Path (Join-Path (Split-Path $_ -Parent) 'dependencies.bicep') # Currently we only need to consider files that have ResourceGroup resources
+        } | ForEach-Object { 
+            $_.Replace('\', '/') 
         }
 
-        # Add additional tokens provided by the user
-        $tokenConfiguration.Tokens += $Tokens
-
-        # Invoke Token Replacement Functionality and Convert Tokens in Parameter Files
-        $null = Convert-TokensInFileList @tokenConfiguration
-
-        try {
-
-            # Extract Parameter Names
-            # -----------------------
-            $storageAccountFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.Storage/storageAccounts' }
-            $storageAccountNames = $storageAccountFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $keyVaultFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.KeyVault/vaults' }
-            $keyVaultNames = $keyVaultFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
-
-            $acrFiles = $parameterFiles | Where-Object { $_ -match 'Microsoft.ContainerRegistry/registries' }
-            $acrNames = $acrFiles | ForEach-Object { (ConvertFrom-Json (Get-Content $_ -Raw)).parameters.name.value } | Where-Object { $null -ne $_ }
+        foreach ($parameterFile in $parameterFiles) {
+
+            foreach ($relevantResourceType in $relevantResourceTypes) {
+                switch ($relevantResourceType) {
+                    'Microsoft.Storage/storageAccounts' {
+                        $filter = 'storageAccountName:'
+                    }
+                    'Microsoft.ContainerRegistry/registries' {
+                        $filter = 'registryName:'
+                    }
+                    'Microsoft.KeyVault/vaults' {
+                        $filter = 'keyVaultName:'
+                    }
+                    Default {
+                        Write-Warning "The entered resource Type '$_' could not be matched."
+                        continue
+                    }
+                }
+            
+                $temp = $null
 
-            $subscriptionId = (Get-AzContext).Subscription.Id
+                # determine if entry is of one of the resourceTypes using the filter variable
+                $temp = Get-Content -Path $parameterFile | ForEach-Object {
```

Comment:
The content of the file should be loaded once in the loop and not for each search (i.e. resource name, serviceshort, etc.)

Reply:
@AlexanderSehr good point.
@fblix I made some changes in your branch.